xonsh/events.py

```diff
@@ -49,37 +63,41 @@ def __call__(self, handler):
         rtn : callable
             The handler
         """
-        #  Using Pythons "private" munging to minimize hypothetical collisions
+        #  Using Python's "private" munging to minimize hypothetical collisions
         handler.__validator = None
+        if debug_level():
+            if not has_kwargs(handler):
```

Comment:
It's not required to run, and does technically slow things down. And it's not like end-users could do something about it.